bionic/cache.py

```diff
@@ -920,6 +920,13 @@ def from_computation(
         else:
             nonfunctional_code_dict['bytecode_hash'] = bytecode_hash
 
+        # Make the functional hash volatile by adding some noise.
+        # This exists for making the provenance unstable when the entity
+        # changes per run and does not persist so that the children won't
+        # use their cached value.
+        if is_functionally_volatile:
```

Comment:
Yup, we can definitely setup the provenance multiple times as long as there is an error after setting it up in `EntityDeriver._complete_task_state()`. However, I don't think anyone would use the provenance if the method threw an exception so this should not be a problem. 

If we really want this to be stable in case a change in future breaks this, we can assign a uuid to the flow object and use it instead of random uuid. Wdyt?

Reply:
I like the idea of attaching a UUID to the flow object. If it's not a lot of work I would lean towards doing that -- otherwise I agree that it's currently hard to see how this would break.